store/store.go

```diff
@@ -501,6 +509,40 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// try advance heighest writeHead based on passed or already written headers.
+func (s *Store[H]) tryAdvanceHead(ctx context.Context, headers ...H) {
+	writeHead := s.writeHead.Load()
+	if writeHead == nil || len(headers) == 0 {
+		return
+	}
+
+	currHeight := (*writeHead).Height()
+
+	// advance based on passed headers.
+	for i := 0; i < len(headers); i++ {
+		if headers[i].Height() != currHeight+1 {
+			break
+		}
+		newHead := headers[i]
+		s.writeHead.Store(&newHead)
+		currHeight++
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
```

Comment:
Giving 1sec to advance, should be enough in most cases. Anyway, we will advance head in next `Append` call.

Reply:
This param needs to be benchmarked to justify. I would opt for a more generous timeout until we see another reason for it.

Reply:
Agree, set 10sec for now and TODO.